fix to set custom headers

t0chk opened the pull request:
Allow to set custom headers in up-level axios options

## Summary
<!-- Add a brief description of the pr: -->

## Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->

t0chk committed:
fix to set custom headers

Allow to set custom headers in up-level axios options

will2022 reviewed (changes requested):
I would probably just modify the `headers` object of `networkOptions`, so it'll become
```js
networkOptions.headers = {
  ...networkOptions.headers,
  'x-referer': APIID,
}
```
and then remove the other headers from the `globalRequestOptions`

t0chk committed:
Custom headers fix

Allow to set custom headers in up-level axios options.
Recoded by @will2022 comment

tiagosiebler commented:
As discussed in telegram, bundled it together with #343
